phd-tests/tests/src/boot_order.rs

```diff
@@ -0,0 +1,533 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::{bail, Error};
+use phd_framework::{
+    disk::{fat::FatFilesystem, DiskSource},
+    test_vm::{DiskBackend, DiskInterface},
+};
+use phd_testcase::*;
+use std::io::Cursor;
+use tracing::warn;
+
+mod efi_utils;
+
+use efi_utils::{
+    bootvar, discover_boot_option_numbers, efipath, find_option_in_boot_order,
+    read_efivar, remove_boot_entry, write_efivar, EfiLoadOption,
+    BOOT_CURRENT_VAR, BOOT_ORDER_VAR, EDK2_EFI_SHELL_GUID,
+    EDK2_FIRMWARE_VOL_GUID, EDK2_UI_APP_GUID,
+};
+
+pub(crate) async fn run_long_command(
+    vm: &phd_framework::TestVm,
+    cmd: &str,
+) -> Result<String, Error> {
+    // Ok, this is a bit whacky: something about the line wrapping for long
+    // commands causes `run_shell_command` to hang instead of ever actually
+    // seeing a response prompt.
```

Comment:
aha, that's a good approach too. so, that works because no individual line is long enough that we're never getting an unexpected `\n` back from the other end. it looks like the shell script also has no individual terribly-long lines, so that works.

i think this workaround is something other tests _could_ use, but i'd rather the serial listener so that it's not needed (destined to become a new issue in a bit here)